tests/test_body_match.js

```diff
@@ -48,6 +49,28 @@ test('match body with regex', function (t) {
 
 });
 
+test('match body (with space character) with regex', function (t) {
+
+  nock('http://encodingsareus.com')
+    .post('/', /a bc/)
+    .reply(200);
+
+  mikealRequest({
+    url: 'http://encodingsareus.com/',
+    method: 'post',
+    json: {
+      auth: {
+        passwd: 'a bc'
+      }
+    },
+  }, function(err, res) {
+    if (err) throw err;
+    assert.equal(res.statusCode, 200);
+    t.end();
+  });
+
+});
```

Comment:
Could you please add another test where the request body is a string to cover https://github.com/node-nock/nock/pull/1089/files#diff-f7a7f1f3d9e28ef78a493002d2cb1f73R14?

Otherwise it’s great, thanks a lot!

Reply:
AFAIK the body is usually a string at the stage. The request library and most http library, stringify the body before sending and then some turn it to buffer.

Reply:
In that case, do we need `return qs.stringify(body).match(spec);`? If all other tests pass without it I’d rather leave it out?

Also I think I’d move the regex check below

```js
  if (Buffer.isBuffer(body)) {
    body = body.toString();
  }
```

What do you think?

Reply:
Hi @gr2m ,

I am sorry for the silence.

> Also I think I’d move the regex check below

Yes, I think it'd make more sense.

I will modify and see if the tests passes